examples/Multiclass_classification_for_transactions.ipynb

```diff
@@ -0,0 +1,2201 @@
+    "                            prompt=prompt,\n",
+    "                            temperature=0,\n",
+    "                            max_tokens=5,\n",
+    "                            top_p=1,\n",
+    "                            frequency_penalty=0,\n",
+    "                            presence_penalty=0,\n",
+    "                            model=COMPLETIONS_MODEL\n",
+    "                            )\n",
+    "        \n",
+    "    return completion_response\n",
+    "\n",
+    "def classify_transaction(transaction,prompt):\n",
+    "    \n",
+    "    prompt = prompt.replace('SUPPLIER_NAME',transaction['Supplier'])\n",
+    "    prompt = prompt.replace('DESCRIPTION_TEXT',transaction['Description'])\n",
+    "    prompt = prompt.replace('TRANSACTION_VALUE',str(transaction['Transaction value (£)']))\n",
+    "    \n",
+    "    classification = request_completion(prompt)['choices'][0]['text'].replace('\\n','')\n",
+    "    \n",
+    "    return classification\n",
+    "\n",
+    "import json\n",
+    "\n",
+    "def check_finetune_classes(train_file,valid_file):\n",
+    "\n",
+    "    train_classes = set()\n",
+    "    valid_classes = set()\n",
+    "    with open(train_file, 'r') as json_file:\n",
+    "        json_list = list(json_file)\n",
+    "        print(len(json_list))\n",
+    "\n",
+    "    for json_str in json_list:\n",
+    "        result = json.loads(json_str)\n",
+    "        train_classes.add(result['completion'])\n",
+    "        #print(f\"result: {result['completion']}\")\n",
+    "        #print(isinstance(result, dict))\n",
+    "\n",
+    "    with open(valid_file, 'r') as json_file:\n",
+    "        json_list = list(json_file)\n",
+    "        print(len(json_list))\n",
+    "\n",
+    "    for json_str in json_list:\n",
+    "        result = json.loads(json_str)\n",
+    "        valid_classes.add(result['completion'])\n",
+    "        #print(f\"result: {result['completion']}\")\n",
+    "        #print(isinstance(result, dict))\n",
+    "        \n",
+    "    if len(train_classes) == len(valid_classes):\n",
+    "        print('All good')\n",
+    "        \n",
+    "    else:\n",
+    "        print('Classes do not match, please prepare data again')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Zero-shot Classification\n",
+    "\n",
+    "We'll first assess the performance of the base models at classifying these transactions using a simple prompt. We'll provide the model with 5 categories and a catch-all of \"Could not classify\" for ones that it cannot place."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 277,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zero_shot_prompt = '''You are a data expert working for the National Library of Scotland. \n",
+    "                      You are analysing all transactions over £25,000 in value and classifying them into one of five categories.\n",
```

Comment:
You have a bunch of whitespace in every line. You should start at the beginning of line (not idented) for every line of the multiline string.